aten/src/ATen/native/cuda/ReduceOpsKernel.cu

```diff
@@ -106,9 +106,13 @@ static void prod_kernel_cuda(TensorIterator& iter) {
 static void mean_kernel_cuda(TensorIterator& iter) {
   if (iter.dtype() == kHalf) {
     return mean_kernel_impl<at::Half, float>(iter);
+  } else if (iter.dtype() == kBFloat16) {
+    return mean_kernel_impl<at::BFloat16, float>(iter);
   } else if (iter.dtype(1) == kHalf && iter.dtype() == kFloat) {
     // type promotion that does cast and reduction in a single kernel
     return mean_kernel_impl<at::Half, float, float>(iter);
+  } else if (iter.dtype(1) == kBFloat16 && iter.dtype() == kBFloat16) {
+    return mean_kernel_impl<at::BFloat16, float, float>(iter);
   }
   AT_DISPATCH_ALL_TYPES_AND(kBFloat16,	iter.dtype(), "mean_cuda", [&]() {
```

Comment:
I have earlier added bfloat16 type here, by doing accumulation also in bfloat16. Now that, you have added the specialization above, this can be changed to `AT_DISPATCH_ALL_TYPES(`